src/zzz_od/gui/dialog/redemption_code_setting_dialog.py

```
@@ -0,0 +1,256 @@
+
+
+class CodeCard(MultiLineSettingCard):
+    """单个兑换码的卡片，类似体力计划卡片"""
+
+    changed = Signal(str, str, int)  # old_code, new_code, end_dt
+    deleted = Signal(str)  # code
+
+    def __init__(self, code: str, end_dt: int, is_new: bool = False, readonly: bool = False, parent=None):
+        self.original_code = code
+        self.is_new = is_new  # 是否是新增的空卡片
+        self.readonly = readonly  # 是否只读（来自 sample 配置）
```

Ruff (0.14.14)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)